save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,267 @@
+                                                "",
+                                                "User name",
+                                            ) {
+                                                changeFields(InputTypes.USER_NAME, it)
+                                                setState {
+                                                    conflictErrorMessage = null
+                                                }
+                                            }
+                                        }
+                                        ReactHTML.button {
+                                            type = ButtonType.button
+                                            className = ClassName("btn btn-info mt-4 mr-3")
+                                            +"Registration"
+                                            onClick = {
+                                                saveUser()
+                                            }
+                                        }
+                                        state.conflictErrorMessage?.let {
+                                            ReactHTML.div {
+                                                className = ClassName("invalid-feedback d-block")
+                                                +it
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        } else if (state.userInfo?.isActive == true) {
+            window.location.href = "#/${FrontendRoutes.PROJECTS.path}"
+        }
+    }
+
+    private fun postImageUpload(element: HTMLInputElement) =
+            scope.launch {
+                setState {
+                    isUploading = true
+                }
+                element.files!!.asList().single().let { file ->
+                    val response: ImageInfo? = post(
+                        "$apiUrl/image/upload?owner=${props.userName}&type=${AvatarType.USER}",
+                        Headers(),
+                        FormData().apply {
+                            append("file", file)
+                        },
+                        loadingHandler = ::classLoadingHandler,
+                    )
+                        .decodeFromJsonString()
+                    setState {
+                        image = response
+                    }
+                }
+                setState {
+                    isUploading = false
+                }
+            }
+
+    /* companion object :
+        RStatics<RegistrationProps, RegistrationViewState, RegistrationView, Context<RequestStatusContext>>(
+            RegistrationView::class
+        ) {
+        init {
+            RegistrationView.contextType = requestStatusContext
+        }
+    }*/
```

Comment:
Why is this companion object commented?

Reply:
It's strange that the Diktat does not swear at this

Reply:
Created issue https://github.com/saveourtool/diktat/issues/1510

Reply:
Why did you completely remove companion object? Don't we need it here? I thought you might want to process responses using `responseHandlerWithValidation`.

Reply:
I think it's unnecessary here. The only check required is to display the response in the Input form itself.